honeybadgermpc/hbavss.py

```diff
@@ -182,3 +191,146 @@ def _get_dealer_msg(self, value):
+            if avss_msg == HbAVSSMessageType.OK and sender not in ok_set:
+                oks_recvd += 1
+                ok_set.add(sender)
+            elif avss_msg == HbAVSSMessageType.READY and sender not in ready_set:
+                readys_recvd += 1
+                ready_set.add(sender)
+
+        multicast(HbAVSSMessageType.READY)
+
+        # if 2t+1 ready -> output shares
+        while readys_recvd < 2 * self.t + 1:
+            sender, avss_msg = await recv()  # First value is the `sid`
+            if avss_msg == HbAVSSMessageType.READY and sender not in ready_set:
+                readys_recvd += 1
+                ready_set.add(sender)
+
+        return shares
+
+    def _get_dealer_msg(self, values):
+        # Sample a random degree-(t,t) bivariate polynomial φ(·,·)
+        # such that each φ(0,k) = sk and φ(i,k) is Pi’s share of sk
+        secret_size = len(values)
+        phi = [None] * secret_size
+        commitments = [None] * secret_size
+        aux_poly = [None] * secret_size
+        # for k ∈ [t+1]
+        #   Ck, auxk <- PolyCommit(SP,φ(·,k))
+        for k in range(secret_size):
+            phi[k] = self.poly.random(self.t, values[k])
+            commitments[k], aux_poly[k] = self.poly_commit.commit(phi[k])
+
+        ephemeral_secret_key = self.field.random()
+        ephemeral_public_key = pow(self.g, ephemeral_secret_key)
+        # for each party Pi and each k ∈ [t+1]
+        #   1. w[i][k] <- CreateWitnesss(Ck,auxk,i)
+        #   2. z[i][k] <- EncPKi(φ(i,k), w[i][k])
+        z = [None] * self.n * secret_size
+        for i in range(self.n):
+            shared_key = pow(self.public_keys[i], ephemeral_secret_key)
+            for k in range(secret_size):
+                witness = self.poly_commit.create_witness(aux_poly[k], i+1)
+                z[i * secret_size + k] = SymmetricCrypto.encrypt(
+                    str(shared_key).encode("utf-8"), (phi[k](i+1), witness))
+
+        return dumps((commitments, ephemeral_public_key, z))
+
+    async def avss(self, avss_id, values=None, dealer_id=None, client_mode=False):
+        """
+        A batched version of avss similar to the one in light version
+        """
+        # If `values` is passed then the node is a 'Sender'
+        # `dealer_id` must be equal to `self.my_id`
+        if values is not None:
+            if dealer_id is None:
+                dealer_id = self.my_id
+            assert dealer_id == self.my_id, "Only dealer can share values."
+        # If `values` is not passed then the node is a 'Recipient'
+        # Verify that the `dealer_id` is not the same as `self.my_id`
+        elif dealer_id is not None:
+            assert dealer_id != self.my_id
+        if client_mode:
+            assert dealer_id is not None
+            assert dealer_id == self.n
+        assert type(avss_id) is int
+
+        logger.debug("[%d] Starting AVSS. Id: %s, Dealer Id: %d, Client Mode: %s",
```

Comment:
Can you update the log to say "Starting Batch AVSS"?

Reply:
Resolved.